### PR TITLE
PT-310 | Add SMS notification

### PR DESCRIPTION
**Pull request by quyenlq**
Add sms notification type, user can select either email or sms or both to receive the notification.
Notification template id will be different for each type of notification
In the future when notification service support send notification by template id then this need to be modified, we probably don't event need `django-ilmoitin` anymore

### CODE REVIEW

File: occurrences/utils.py [hunk 1]
```diff
@@ -1,18 +1,100 @@
...
+    if not template:
+        logger.warning(
+            'No notification template created for "{}" event, '
+            "not sending anything.".format(notification_template_id)
+        )
+        return
+
+    try:
+        _, _, body_text = render_notification_template(template, context, language)
+    except NotificationTemplate.DoesNotExist:
```
[the remark this answers is not in the document]
Comment (quyenlq) [reply occurrences/utils.py:79]: I think the `render_notification_template` will throw that exception if the template translation corresponds to `language` is missing.

File: occurrences/notifications.py [hunk 2]
```diff
@@ -14,6 +14,12 @@
 notifications.register(
     NotificationType.OCCURRENCE_UNENROLMENT, _("occurrence unenrolment")
 )
+notifications.register(
+    NotificationType.OCCURRENCE_ENROLMENT_SMS, _("occurrence " "enrolment sms")
```
[the remark this answers is not in the document]
Comment (quyenlq) [reply occurrences/notifications.py:18]: Seems like my IDE try to wrap the line while I press the auto-indent hotkey, then `black` reformatted it again 😅

File: occurrences/utils.py [hunk 3]
```diff
@@ -1,18 +1,100 @@
...
+        ]
+        send_sms_notification(
+            destinations,
+            context,
+            notification_sms_template_id,
+            study_group.person.language,
+        )
+
+
+def send_sms_notification(destinations, context, notification_template_id, language):
```
[the remark this answers is not in the document]
Comment (quyenlq) [reply occurrences/utils.py:58]: Good point! Changed that

**Comment by quyenlq** [comment]
> Looks good!
> 
> One general observation, using the words "notification type" for two things (notification template type and notification "transport method" (email / SMS)) makes things pretty confusing at times, maybe one of those could be renamed somehow?

Yeah, it indeed confused me while implementing the feature. I tried to replace `notification template type` to `notification_template_id` at some places. I think I can improve it a bit more.

File: occurrences/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tuomas777) [reply occurrences/utils.py:83]: Ah ok! I think we could check at some point if this could be improved in ilmoitin.

**Pull request merged**